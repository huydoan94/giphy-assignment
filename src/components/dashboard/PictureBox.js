import React from 'react';
import { compose } from 'redux';
import { string, shape, func } from 'prop-types';
import {
  Card, CardActionArea, CardMedia, CardContent, CardHeader,
  Typography, Avatar,
} from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import moment from 'moment';

import { withPictureFullscreenDialog } from './PictureBox.hoc';
import styles from './PictureBox.styles';

export function PictureBox({
  title, fixedHeightUrl, trendingDate,
  username, userAvatarUrl, classes, handleOpen,
}) {
  return (
    <Card>
      <CardActionArea onClick={handleOpen}>
        <CardMedia
          className={classes.media}
          image={fixedHeightUrl}
          title={title}
        />
        <CardContent>
          <Typography gutterBottom variant="h5" component="h2" className={classes.pictureTitle}>
            {title}
          </Typography>
        </CardContent>
        <CardHeader
          avatar={<Avatar src={userAvatarUrl}>UU</Avatar>}
          title={username}
          subheader={moment(trendingDate).format('l LT')}
        />
      </CardActionArea>
    </Card>
  );
}

PictureBox.propTypes = {
  title: string.isRequired,
  fixedHeightUrl: string.isRequired,
  trendingDate: string.isRequired,
  username: string.isRequired,
  userAvatarUrl: string,
  handleOpen: func.isRequired,
  classes: shape({
    media: string,
    pictureFooter: string,
  }).isRequired,
};

PictureBox.defaultProps = {
  userAvatarUrl: null,
};

export default compose(
  withStyles(styles),
  withPictureFullscreenDialog,
)(PictureBox);
