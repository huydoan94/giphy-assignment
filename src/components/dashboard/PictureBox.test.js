import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import {
  CardActionArea,
} from '@material-ui/core';

import { PictureBox } from './PictureBox';

describe('<PictureBox />', () => {
  let props;
  let component;

  const createComponent = () => {
    component = mount(<PictureBox {...props} />);
  };

  beforeEach(() => {
    props = {
      title: 'title',
      fixedHeightUrl: 'fixed_height_url',
      trendingDate: '2019-08-27T03:04:57+07:00',
      username: 'username',
      userAvatarUrl: 'avatar_url',
      handleOpen: jest.fn(),
      classes: {},
    };
    createComponent();
  });

  it('should match snapshot', () => {
    expect(toJSON(component)).toMatchSnapshot();
  });

  it('CardActionArea click should call handleOpen', () => {
    component.find(CardActionArea).simulate('click');
    expect(props.handleOpen).toHaveBeenCalledTimes(1);
  });
});
