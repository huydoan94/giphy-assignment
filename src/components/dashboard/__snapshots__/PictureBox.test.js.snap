// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PictureBox /> should match snapshot 1`] = `
<PictureBox
  classes={Object {}}
  fixedHeightUrl="fixed_height_url"
  handleOpen={[Function]}
  title="title"
  trendingDate="2019-08-27T03:04:57+07:00"
  userAvatarUrl="avatar_url"
  username="username"
>
  <WithStyles(ForwardRef(Card))>
    <ForwardRef(Card)
      classes={
        Object {
          "root": "MuiCard-root",
        }
      }
    >
      <WithStyles(ForwardRef(Paper))
        className="MuiCard-root"
        elevation={1}
      >
        <ForwardRef(Paper)
          className="MuiCard-root"
          classes={
            Object {
              "elevation0": "MuiPaper-elevation0",
              "elevation1": "MuiPaper-elevation1",
              "elevation10": "MuiPaper-elevation10",
              "elevation11": "MuiPaper-elevation11",
              "elevation12": "MuiPaper-elevation12",
              "elevation13": "MuiPaper-elevation13",
              "elevation14": "MuiPaper-elevation14",
              "elevation15": "MuiPaper-elevation15",
              "elevation16": "MuiPaper-elevation16",
              "elevation17": "MuiPaper-elevation17",
              "elevation18": "MuiPaper-elevation18",
              "elevation19": "MuiPaper-elevation19",
              "elevation2": "MuiPaper-elevation2",
              "elevation20": "MuiPaper-elevation20",
              "elevation21": "MuiPaper-elevation21",
              "elevation22": "MuiPaper-elevation22",
              "elevation23": "MuiPaper-elevation23",
              "elevation24": "MuiPaper-elevation24",
              "elevation3": "MuiPaper-elevation3",
              "elevation4": "MuiPaper-elevation4",
              "elevation5": "MuiPaper-elevation5",
              "elevation6": "MuiPaper-elevation6",
              "elevation7": "MuiPaper-elevation7",
              "elevation8": "MuiPaper-elevation8",
              "elevation9": "MuiPaper-elevation9",
              "root": "MuiPaper-root",
              "rounded": "MuiPaper-rounded",
            }
          }
          elevation={1}
        >
          <div
            className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
          >
            <WithStyles(ForwardRef(CardActionArea))
              onClick={[Function]}
            >
              <ForwardRef(CardActionArea)
                classes={
                  Object {
                    "focusHighlight": "MuiCardActionArea-focusHighlight",
                    "focusVisible": "Mui-focusVisible",
                    "root": "MuiCardActionArea-root",
                  }
                }
                onClick={[Function]}
              >
                <WithStyles(ForwardRef(ButtonBase))
                  className="MuiCardActionArea-root"
                  focusVisibleClassName="Mui-focusVisible"
                  onClick={[Function]}
                >
                  <ForwardRef(ButtonBase)
                    className="MuiCardActionArea-root"
                    classes={
                      Object {
                        "disabled": "Mui-disabled",
                        "focusVisible": "Mui-focusVisible",
                        "root": "MuiButtonBase-root",
                      }
                    }
                    focusVisibleClassName="Mui-focusVisible"
                    onClick={[Function]}
                  >
                    <button
                      className="MuiButtonBase-root MuiCardActionArea-root"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <WithStyles(ForwardRef(CardMedia))
                        image="fixed_height_url"
                        title="title"
                      >
                        <ForwardRef(CardMedia)
                          classes={
                            Object {
                              "media": "MuiCardMedia-media",
                              "root": "MuiCardMedia-root",
                            }
                          }
                          image="fixed_height_url"
                          title="title"
                        >
                          <div
                            className="MuiCardMedia-root"
                            style={
                              Object {
                                "backgroundImage": "url(\\"fixed_height_url\\")",
                              }
                            }
                            title="title"
                          />
                        </ForwardRef(CardMedia)>
                      </WithStyles(ForwardRef(CardMedia))>
                      <WithStyles(ForwardRef(CardContent))>
                        <ForwardRef(CardContent)
                          classes={
                            Object {
                              "root": "MuiCardContent-root",
                            }
                          }
                        >
                          <div
                            className="MuiCardContent-root"
                          >
                            <WithStyles(ForwardRef(Typography))
                              component="h2"
                              gutterBottom={true}
                              variant="h5"
                            >
                              <ForwardRef(Typography)
                                classes={
                                  Object {
                                    "alignCenter": "MuiTypography-alignCenter",
                                    "alignJustify": "MuiTypography-alignJustify",
                                    "alignLeft": "MuiTypography-alignLeft",
                                    "alignRight": "MuiTypography-alignRight",
                                    "body1": "MuiTypography-body1",
                                    "body2": "MuiTypography-body2",
                                    "button": "MuiTypography-button",
                                    "caption": "MuiTypography-caption",
                                    "colorError": "MuiTypography-colorError",
                                    "colorInherit": "MuiTypography-colorInherit",
                                    "colorPrimary": "MuiTypography-colorPrimary",
                                    "colorSecondary": "MuiTypography-colorSecondary",
                                    "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                    "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                    "displayBlock": "MuiTypography-displayBlock",
                                    "displayInline": "MuiTypography-displayInline",
                                    "gutterBottom": "MuiTypography-gutterBottom",
                                    "h1": "MuiTypography-h1",
                                    "h2": "MuiTypography-h2",
                                    "h3": "MuiTypography-h3",
                                    "h4": "MuiTypography-h4",
                                    "h5": "MuiTypography-h5",
                                    "h6": "MuiTypography-h6",
                                    "noWrap": "MuiTypography-noWrap",
                                    "overline": "MuiTypography-overline",
                                    "paragraph": "MuiTypography-paragraph",
                                    "root": "MuiTypography-root",
                                    "srOnly": "MuiTypography-srOnly",
                                    "subtitle1": "MuiTypography-subtitle1",
                                    "subtitle2": "MuiTypography-subtitle2",
                                  }
                                }
                                component="h2"
                                gutterBottom={true}
                                variant="h5"
                              >
                                <h2
                                  className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
                                >
                                  title
                                </h2>
                              </ForwardRef(Typography)>
                            </WithStyles(ForwardRef(Typography))>
                          </div>
                        </ForwardRef(CardContent)>
                      </WithStyles(ForwardRef(CardContent))>
                      <WithStyles(ForwardRef(CardHeader))
                        avatar={
                          <Unknown
                            src="avatar_url"
                          >
                            UU
                          </Unknown>
                        }
                        subheader="8/27/2019 3:04 AM"
                        title="username"
                      >
                        <ForwardRef(CardHeader)
                          avatar={
                            <Unknown
                              src="avatar_url"
                            >
                              UU
                            </Unknown>
                          }
                          classes={
                            Object {
                              "action": "MuiCardHeader-action",
                              "avatar": "MuiCardHeader-avatar",
                              "content": "MuiCardHeader-content",
                              "root": "MuiCardHeader-root",
                              "subheader": "MuiCardHeader-subheader",
                              "title": "MuiCardHeader-title",
                            }
                          }
                          subheader="8/27/2019 3:04 AM"
                          title="username"
                        >
                          <div
                            className="MuiCardHeader-root"
                          >
                            <div
                              className="MuiCardHeader-avatar"
                            >
                              <WithStyles(ForwardRef(Avatar))
                                src="avatar_url"
                              >
                                <ForwardRef(Avatar)
                                  classes={
                                    Object {
                                      "colorDefault": "MuiAvatar-colorDefault",
                                      "img": "MuiAvatar-img",
                                      "root": "MuiAvatar-root",
                                    }
                                  }
                                  src="avatar_url"
                                >
                                  <div
                                    className="MuiAvatar-root"
                                  >
                                    <img
                                      className="MuiAvatar-img"
                                      src="avatar_url"
                                    />
                                  </div>
                                </ForwardRef(Avatar)>
                              </WithStyles(ForwardRef(Avatar))>
                            </div>
                            <div
                              className="MuiCardHeader-content"
                            >
                              <WithStyles(ForwardRef(Typography))
                                className="MuiCardHeader-title"
                                component="span"
                                display="block"
                                variant="body2"
                              >
                                <ForwardRef(Typography)
                                  className="MuiCardHeader-title"
                                  classes={
                                    Object {
                                      "alignCenter": "MuiTypography-alignCenter",
                                      "alignJustify": "MuiTypography-alignJustify",
                                      "alignLeft": "MuiTypography-alignLeft",
                                      "alignRight": "MuiTypography-alignRight",
                                      "body1": "MuiTypography-body1",
                                      "body2": "MuiTypography-body2",
                                      "button": "MuiTypography-button",
                                      "caption": "MuiTypography-caption",
                                      "colorError": "MuiTypography-colorError",
                                      "colorInherit": "MuiTypography-colorInherit",
                                      "colorPrimary": "MuiTypography-colorPrimary",
                                      "colorSecondary": "MuiTypography-colorSecondary",
                                      "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                      "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                      "displayBlock": "MuiTypography-displayBlock",
                                      "displayInline": "MuiTypography-displayInline",
                                      "gutterBottom": "MuiTypography-gutterBottom",
                                      "h1": "MuiTypography-h1",
                                      "h2": "MuiTypography-h2",
                                      "h3": "MuiTypography-h3",
                                      "h4": "MuiTypography-h4",
                                      "h5": "MuiTypography-h5",
                                      "h6": "MuiTypography-h6",
                                      "noWrap": "MuiTypography-noWrap",
                                      "overline": "MuiTypography-overline",
                                      "paragraph": "MuiTypography-paragraph",
                                      "root": "MuiTypography-root",
                                      "srOnly": "MuiTypography-srOnly",
                                      "subtitle1": "MuiTypography-subtitle1",
                                      "subtitle2": "MuiTypography-subtitle2",
                                    }
                                  }
                                  component="span"
                                  display="block"
                                  variant="body2"
                                >
                                  <span
                                    className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
                                  >
                                    username
                                  </span>
                                </ForwardRef(Typography)>
                              </WithStyles(ForwardRef(Typography))>
                              <WithStyles(ForwardRef(Typography))
                                className="MuiCardHeader-subheader"
                                color="textSecondary"
                                component="span"
                                display="block"
                                variant="body2"
                              >
                                <ForwardRef(Typography)
                                  className="MuiCardHeader-subheader"
                                  classes={
                                    Object {
                                      "alignCenter": "MuiTypography-alignCenter",
                                      "alignJustify": "MuiTypography-alignJustify",
                                      "alignLeft": "MuiTypography-alignLeft",
                                      "alignRight": "MuiTypography-alignRight",
                                      "body1": "MuiTypography-body1",
                                      "body2": "MuiTypography-body2",
                                      "button": "MuiTypography-button",
                                      "caption": "MuiTypography-caption",
                                      "colorError": "MuiTypography-colorError",
                                      "colorInherit": "MuiTypography-colorInherit",
                                      "colorPrimary": "MuiTypography-colorPrimary",
                                      "colorSecondary": "MuiTypography-colorSecondary",
                                      "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                      "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                      "displayBlock": "MuiTypography-displayBlock",
                                      "displayInline": "MuiTypography-displayInline",
                                      "gutterBottom": "MuiTypography-gutterBottom",
                                      "h1": "MuiTypography-h1",
                                      "h2": "MuiTypography-h2",
                                      "h3": "MuiTypography-h3",
                                      "h4": "MuiTypography-h4",
                                      "h5": "MuiTypography-h5",
                                      "h6": "MuiTypography-h6",
                                      "noWrap": "MuiTypography-noWrap",
                                      "overline": "MuiTypography-overline",
                                      "paragraph": "MuiTypography-paragraph",
                                      "root": "MuiTypography-root",
                                      "srOnly": "MuiTypography-srOnly",
                                      "subtitle1": "MuiTypography-subtitle1",
                                      "subtitle2": "MuiTypography-subtitle2",
                                    }
                                  }
                                  color="textSecondary"
                                  component="span"
                                  display="block"
                                  variant="body2"
                                >
                                  <span
                                    className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                                  >
                                    8/27/2019 3:04 AM
                                  </span>
                                </ForwardRef(Typography)>
                              </WithStyles(ForwardRef(Typography))>
                            </div>
                          </div>
                        </ForwardRef(CardHeader)>
                      </WithStyles(ForwardRef(CardHeader))>
                      <span
                        className="MuiCardActionArea-focusHighlight"
                      />
                      <NoSsr>
                        <WithStyles(undefined)
                          center={false}
                        >
                          <ForwardRef(TouchRipple)
                            center={false}
                            classes={
                              Object {
                                "child": "MuiTouchRipple-child",
                                "childLeaving": "MuiTouchRipple-childLeaving",
                                "childPulsate": "MuiTouchRipple-childPulsate",
                                "ripple": "MuiTouchRipple-ripple",
                                "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                "rippleVisible": "MuiTouchRipple-rippleVisible",
                                "root": "MuiTouchRipple-root",
                              }
                            }
                          >
                            <span
                              className="MuiTouchRipple-root"
                            >
                              <TransitionGroup
                                childFactory={[Function]}
                                component={null}
                                exit={true}
                              />
                            </span>
                          </ForwardRef(TouchRipple)>
                        </WithStyles(undefined)>
                      </NoSsr>
                    </button>
                  </ForwardRef(ButtonBase)>
                </WithStyles(ForwardRef(ButtonBase))>
              </ForwardRef(CardActionArea)>
            </WithStyles(ForwardRef(CardActionArea))>
          </div>
        </ForwardRef(Paper)>
      </WithStyles(ForwardRef(Paper))>
    </ForwardRef(Card)>
  </WithStyles(ForwardRef(Card))>
</PictureBox>
`;
